pkg/controller/signup.go

```diff
@@ -28,7 +29,8 @@ func NewSignup(config *configuration.Registry, signupService signup.Service) *Si
 
 // PostHandler creates a Signup resource
 func (s *Signup) PostHandler(ctx *gin.Context) {
-	userSignup, err := s.signupService.CreateUserSignup(ctx.GetString(context.UsernameKey), ctx.GetString(context.SubKey))
+	compliantEmailLabel := toolchain.ToValidValue(ctx.GetString(context.EmailKey))
```

Comment:
It's not a label. It's annotation.
And what is `toolchain.ToValidValue()`? I don't see it in common.

Reply:
@alexeykazakov i changed these PRs to WIP so that i can change them to be labels based off of a discussion with @xcoulon. Apparently there is no way to change a PR to a Draft once it's open =/

Reply:
@alexeykazakov why don't we convert the email to a valid label value and set it as a label on the user signup? that way, we could look-up a usersignup resource based on the email address of the user to approve... WDYT?

Reply:
It would be nice to be able to look up a user by email indeed. But just transforming email to a dns compliant string won’t be enough for that. You can’t tell what was the original email just by looking at the transformed string (if we want to keep it human readable).
And we need the original email anyway. For now it will help us with approving users. Later we will need it for email notifications.
So let’s add this annotation first and deal with the lookup later.